app/src/Clone/CCServer.ts

```diff
@@ -38,38 +38,70 @@ export default class CCServer {
   config: Config;
   isActive = false;
   lastRequestTime = process.hrtime();
+  pendingProblems: ProblemData[] = [];
+  totalProblemsCount = 0;
+  receivedProblemsCount = 0;
+
   constructor(config: Config) {
     this.config = config;
     this.app.use(express.json());
     this.app.post("/", (request, response) => {
+      const problemData: ProblemData = request.body;
+      const batchSize = problemData.batch?.size || 1;
+
+      if (this.totalProblemsCount === 0) {
+        this.totalProblemsCount = batchSize;
+      }
+
+      console.log(`Received problem ${this.receivedProblemsCount + 1} of ${this.totalProblemsCount}`);
+      this.pendingProblems.push(problemData);
+      this.receivedProblemsCount++;
+
       response.writeHead(200, { "Content-Type": "text/html" });
       response.end("OK");
 
-      const problemData: ProblemData = request.body;
+      if (this.receivedProblemsCount === this.totalProblemsCount) {
+        console.log();
+        this.processAllProblems();
+        this.resetCounts();
+      } else {
+        console.log(
+          chalk.red(
+            "Could not parse all problems. Aborting...\n"
```

Comment:
Will this be printed everytime we receive a problem ? since `receivedProblemCount` is not equal to `totalProblemCount` for `N - 1` times